File: OCH_Tao/ABC/B/225.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input.shift(), 10);
let M = new Map();
for (const i of input) {
  const [A, B] = i.split(" ").map(x => parseInt(x, 10));
  M.set(A, (M.get(A) ?? 0) + 1);
  M.set(B, (M.get(B) ?? 0) + 1);
}
console.log(Math.max(...M.values()) === N - 1 && M.size === N ? "Yes" : "No");

File: OCH_Tao/ABC/B/225.py
N = int(input())
D = dict()
for i in range(N-1):
  A,B = map(int,input().split())
  D[A]=D.get(A,0)+1
  D[B]=D.get(B,0)+1
print("Yes" if max(D.values())==N-1 and len(D)==N else "No")